HACKING.md: Clarify installation (#20974)
cockpit-project/cockpit: HACKING.md

## Working on Cockpit's session pages

Most contributors want to work on the web (HTML, JavaScript, CSS) parts of Cockpit.

### Install Cockpit

First, install Cockpit on your local machine as described in:

<https://cockpit-project.org/running.html>

> ### Build session pages

Reviewer: ```suggestion
### Build session pages

```

(added space below the heading)

Next, run this command from your top level Cockpit checkout directory, and make
sure to run it as the same user that you'll use to log into Cockpit below.

    mkdir -p ~/.local/share/
    ln -s $(pwd)/dist ~/.local/share/cockpit

This will cause Cockpit to read JavaScript, HTML, and CSS files directly from the
locally built package output directory instead of using the system-installed Cockpit
files.

> The recommended and fastest way is build your bundles is using the
> "watch" mode (`-w` or`--watch`) on the page that you are working on.
> For example, if you want to work on anything in [pkg/systemd](./pkg/systemd/), run:

Reviewer: ```suggestion
The recommended way to build bundles is to use the "watch" mode
(`-w` or`--watch`) on the page you are working on. For example, if you want to
work on anything in [pkg/systemd](./pkg/systemd/), run:
```

Minor adjustments and a text reflow. (I also don't think we need to say "and fastest", so I removed that.)

    ./build.js -w

> Now you can log into Cockpit on your local Linux machine at the following
> address. Use the same user and password that you used to log into your Linux
> desktop.

Reviewer: ```suggestion
Now you can log into Cockpit on your local Linux machine at the following
address, using the same username and password as your desktop login:
```

<http://localhost:9090>

> Watch mode rebuilds automatically on any additional changes to the source files,
> so you only need to reload Cockpit in your browser after watch finishes rebuilding.
> Press `Ctrl-C` to stop watch mode once you are done with changing the code.

Reviewer: ```suggestion
Watch mode automatically rebuilds when source files are modified. Once it
finishes building, refresh your browser to see the changes in Cockpit.
Press `Ctrl-C` to stop watch mode when you are done changing the code.
```

> ### Testing

Reviewer: ```suggestion
### Testing
 
```

Added blank line

    RSYNC=c ./build.js -w kdump
    RSYNC=c ./build.js -w

> ### Returning to system packages

Reviewer: ```suggestion
### Returning to system packages

```

Added blank line
